### PR TITLE
ApplicationErrorScanner standardize input file location

### PR DESCRIPTION
**Pull request by kingthorin**
Move `application_errors.xml` from resources directory to zapHomeFiles.
Update scan rule and unit test to address the change.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

**Comment by kingthorin** [comment]
I can add a change log note if you think that's needed, I kinda figured it wasn't a user relevant change.

**Comment by thc202** [comment]
Worth adding, so that the user is aware that it can now add/change the errors (might worth mention in the help too?).

**Comment by thc202** [comment]
There's another test that needs to be changed `ApplicationErrorScannerRegexMatcherUnitTest` (Travis CI failed).

File: addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ApplicationErrorScannerUnitTest.java [hunk 1]
```diff
@@ -41,6 +45,24 @@ protected ApplicationErrorScanner createScanner() {
...
+        String content =
+                "<?xml version=\"1.0\" standalone=\"no\"?>\n"
+                        + "<!-- \n"
+                        + "UnitTest File\n"
+                        + "-->\n"
+                        + "<Patterns>\n"
+                        + "  <Pattern type=\"string\">Microsoft OLE DB Provider for ODBC Drivers</Pattern>\n"
+                        + "  <Pattern type=\"regex\">(?i)Line\\s\\d+:\\sIncorrect\\ssyntax\\snear\\s'[^']*'</Pattern>\n"
+                        + "</Patterns>";
+        Files.write(testFile, content.getBytes());
```
Comment (thc202) [inline addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/ApplicationErrorScannerUnitTest.java]: Better to get the bytes with `StandardCharsets.UTF_8` (that's what will be used to read the file).

**Comment by kingthorin** [comment]
All fixed/addressed.

**Comment by thc202** [comment]
Thanks!

**Pull request merged**